src/main/java/org/elasticsearch/client/IndicesAdminClient.java

```diff
@@ -588,6 +588,21 @@
     AnalyzeRequestBuilder prepareAnalyze(String text);
 
     /**
+     * Analyze texts under the provided index.
+     *
+     * @param index The index name
+     * @param text  The array of text to analyze
+     */
+    AnalyzeRequestBuilder prepareAnalyzeWithIndexAndMultiValued(@Nullable String index, String... text);
+
+    /**
+     * Analyze texts.
+     *
+     * @param text The array of text to analyze
+     */
+    AnalyzeRequestBuilder prepareAnalyzeWithMultiValued(String... text);
```

Comment:
> what do you think about dropping all text parameters for I/F simplification and adding the setter of text/texts to AnalyzeRequestBuilder?

that is an option too. For now I'd just expose a `prepareAnalyze()` then with no parameters, leave the existing ones for bw comp, and add the needed setters where needed?